mono/utils/atomic.h

```diff
@@ -214,33 +214,33 @@ static inline void InterlockedWrite16(volatile gint16 *dst, gint16 val)
 #define gcc_sync_fetch_and_add(a, b) __sync_fetch_and_add (a, b)
 #endif
 
-static inline gint32 InterlockedCompareExchange(volatile gint32 *dest,
+static inline gint32 mono_atomic_cas_i32(volatile gint32 *dest,
 						gint32 exch, gint32 comp)
 {
 	return gcc_sync_val_compare_and_swap (dest, comp, exch);
 }
 
-static inline gpointer InterlockedCompareExchangePointer(volatile gpointer *dest, gpointer exch, gpointer comp)
+static inline gpointer mono_atomic_cas_ptr(volatile gpointer *dest, gpointer exch, gpointer comp)
 {
 	return gcc_sync_val_compare_and_swap (dest, comp, exch);
 }
 
-static inline gint32 InterlockedAdd(volatile gint32 *dest, gint32 add)
+static inline gint32 mono_atomic_add_i32(volatile gint32 *dest, gint32 add)
 {
 	return gcc_sync_add_and_fetch (dest, add);
 }
 
-static inline gint32 InterlockedIncrement(volatile gint32 *val)
+static inline gint32 mono_atomic_inc_i32(volatile gint32 *val)
 {
 	return gcc_sync_add_and_fetch (val, 1);
 }
 
-static inline gint32 InterlockedDecrement(volatile gint32 *val)
+static inline gint32 mono_atomic_dec_i32(volatile gint32 *val)
 {
 	return gcc_sync_sub_and_fetch (val, 1);
 }
 
-static inline gint32 InterlockedExchange(volatile gint32 *val, gint32 new_val)
+static inline gint32 mono_atomic_xchg_i32(volatile gint32 *val, gint32 new_val)
```

Comment:
Considering our CAS functions are called `cas` (compare and swap), it would make sense to call these functions `swap`, or alternatively rename `cas` to `cmpxchg`.

It's a minor point though, and it's unlikely anyone would be confused about the names anyway, so feel free to ignore.